pkg/operator/apis/monitoring/v1/monitoring_types.go

```diff
@@ -77,17 +81,17 @@ type MonitoringStatus struct {
 	Conditions []MonitoringCondition `json:"conditions,omitempty"`
 }
 
-// SetMonitoringCondition merges the provided condition if the resource generation changed or there is
-// a status condition state transition.
-func (status *MonitoringStatus) SetMonitoringCondition(gen int64, now metav1.Time, cond *MonitoringCondition) (bool, error) {
+// SetMonitoringCondition merges the provided valid condition if the resource generation changed or
+// there is a status condition state transition.
+func (status *MonitoringStatus) SetMonitoringCondition(gen int64, now metav1.Time, cond *MonitoringCondition) bool {
 	var (
 		specChanged              = status.ObservedGeneration != gen
 		statusTransition, update bool
 		conds                    = make(map[MonitoringConditionType]*MonitoringCondition)
 	)
 
-	if cond.Type == "" || cond.Status == "" {
-		return update, errInvalidCond
+	if !cond.IsValid() {
+		return false
```

Comment:
How often does this happen? Should we log when this happens?

Reply:
In practice, never. We always call `SetMonitoringCondition` with a valid condition.

If someone wants to build a `MonitoringCondition` dynamically and wants logging, they could call `IsValid` and then log.

Reply:
OK!